### DISCUSSION TITLE
v2.3.0-beta.2

### DISCUSSION THREAD

**Discussion by IMBlues** [TencentBlueKing/bk-user#111] [General]
### API

- [NEW] 支持传递参数，可以拉取已软删除数据 [#1](https://github.com/TencentBlueKing/bk-user/issues/1)
- [NEW] 支持恢复已删除的数据 [#15](https://github.com/TencentBlueKing/bk-user/issues/15)
- [NEW] 支持记录 LDAP/AD 同步组织架构/人员信息的结构化日志 [#27](https://github.com/TencentBlueKing/bk-user/issues/27)
- [OPTIMIZATION] 优化数据源同步任务为后台执行 [#32](https://github.com/TencentBlueKing/bk-user/issues/32)
- [FIX] 修正 SettingMeta 默认路径参数为 id [#45](https://github.com/TencentBlueKing/bk-user/issues/45)
- [FIX] 修正 API /api/v2/batch/profiles/ 中 swagger 参数 query_ids 缺失问题 [#26](https://github.com/TencentBlueKing/bk-user/issues/26)

### SaaS

- [NEW] 支持数据源同步任务页面查看 [#32](https://github.com/TencentBlueKing/bk-user/issues/32)
- [NEW] 支持登录日志导出 [#32](https://github.com/TencentBlueKing/bk-user/issues/32)
- [OPTIMIZATION] 文字按钮的风格保持统一 (https://github.com/TencentBlueKing/bk-user/issues/105)
- [FIX] 修改页面拉取上级组件，从全量拉取改为类似【人员选择器】的分页拉取组件 [#55](https://github.com/TencentBlueKing/bk-user/issues/55)
- [FIX] 修复“从其他组织拉取用户”异常问题 (https://github.com/TencentBlueKing/bk-user/issues/102)
- [FIX] 修复数值型自定义字段在页面上输入时没有异常提示 (https://github.com/TencentBlueKing/bk-user/issues/101)

## 二进制部署升级注意事项

部署前需要升级相关 bash 脚本，具体内容需要检查：

在 `release_usermgr.sh` 和 `install_usermgr.sh` 文件中， 如果存在 `export DJANGO_SETTINGS_MODULE=config.ce.prod` 这一行，需要手动将此行删除。

## What's Changed
* Fix: Add sanity check for disabled category in creating profiles by @uddmorningsun in https://github.com/TencentBlueKing/bk-user/pull/94
* fix: skip when certain key missing in extras of legacy data by @IMBlues in https://github.com/TencentBlueKing/bk-user/pull/103
* fix: 文字按钮的风格保持统一 #105 by @yuri0528 in https://github.com/TencentBlueKing/bk-user/pull/108
* fix: 修改拉取用户接口逻辑 #102 by @yuri0528 in https://github.com/TencentBlueKing/bk-user/pull/109
* fix: 修正权限中心注册脚本环境变量 APP_ID & APP_TOKEN -> BK_APP_CODE & BK_APP_SECRET by @IMBlues in https://github.com/TencentBlueKing/bk-user/pull/104
* chore: update version to v2.3.0-beta.2 by @IMBlues in https://github.com/TencentBlueKing/bk-user/pull/110

## New Contributors
* @yuri0528 made their first contribution in https://github.com/TencentBlueKing/bk-user/pull/108

**Full Changelog**: https://github.com/TencentBlueKing/bk-user/compare/v2.3.0-beta.1...v2.3.0-beta.2

<hr /><em>This discussion was created from the release <a href='https://github.com/TencentBlueKing/bk-user/releases/tag/v2.3.0-beta.2'>v2.3.0-beta.2</a>.</em>

**Comment by uddmorningsun** [comment]
> 在 release_usermgr.sh 和 install_usermgr.sh 文件中， 如果存在 export DJANGO_SETTINGS_MODULE=config.ce.prod 这一行，需要手动将此行删除。

这个方式有点 ugly :sweat_smile:，增加了用户的操作复杂度。是否可以在调用 ` release_usermgr.sh 和 install_usermgr.sh` 中的脚本中使用 `sed` 去判断并处理，这块是有其他的逻辑存在导致必须要人为手动参与？

**Reply by IMBlues** [reply]
这里我的描述可能有点歧义，实际上是需要在升级前，对原来的升级脚本内容进行修改。

实际上的背景是原来的代码在启动时，额外需求部署脚本增加 `export DJANGO_SETTINGS_MODULE=config.ce.prod` 这行提供坏境变量，而新的版本中我们优化了一块儿的逻辑，不需要部署脚本额外增加这行了。

这一脚本变更我们也已经向部署侧维护同学提出 issue 了，新版本中的脚本会天然去除这一行添加。如果是旧的部署方案，也没有升级，需要在部署前检查部署脚本，将这行 bash 手动删去
